server/base/src/main/java/org/apache/accumulo/server/util/checkCommand/MetadataCheckRunner.java

```diff
@@ -0,0 +1,177 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.util.checkCommand;
+
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.SortedMap;
+
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.data.TabletId;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.user.WholeRowIterator;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema;
+import org.apache.accumulo.core.security.AuthorizationContainer;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.util.ColumnFQ;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.constraints.MetadataConstraints;
+import org.apache.accumulo.server.constraints.SystemEnvironment;
+import org.apache.accumulo.server.util.Admin;
+import org.apache.hadoop.io.Text;
+import org.apache.zookeeper.KeeperException;
+
+public interface MetadataCheckRunner extends CheckRunner {
+
+  String tableName();
+
+  TableId tableId();
+
+  Set<ColumnFQ> requiredColFQs();
+
+  Set<Text> requiredColFams();
+
+  default String scanning() {
+    return String.format("%s (%s) table", tableName(), tableId());
+  }
+
+  /**
+   * Ensures that the {@link #tableName()} table (either metadata or root table) has all columns
+   * that are expected. For the root metadata, ensures that the expected "columns" exist in ZK.
+   */
+  default Admin.CheckCommand.CheckStatus checkRequiredColumns(ServerContext context,
+      Admin.CheckCommand.CheckStatus status)
+      throws TableNotFoundException, InterruptedException, KeeperException {
+    Set<ColumnFQ> requiredColFQs;
+    Set<Text> requiredColFams;
+    boolean missingReqCol = false;
+
+    System.out.printf("Scanning the %s for missing required columns...\n", scanning());
+    try (Scanner scanner = context.createScanner(tableName(), Authorizations.EMPTY)) {
+      var is = new IteratorSetting(100, "tablets", WholeRowIterator.class);
+      scanner.addScanIterator(is);
```

Comment:
could also configure the scanner to only fetch the required columns.  This will bring back less data, like it will not bring back tablets files.